### PR TITLE
Expose timeout in v1 Parse API

### LINKED ISSUE #78: Set timeout for parse request
I'm running analyzis on many files some of them are big.
I see quite many
```
rpc error: code = DeadlineExceeded desc = context deadline exceeded
```
errors.

I would prefer to process slower but avoid these errors.

Timeout future similar to client-go could help here.

### PR DESCRIPTION
**Pull request by bzz**
Fixes #78 for v1 API.

 - clarifies status of v1 support in docs
 - adds `parseWithTimeout()` method
 - moves all test to the right dir (was a typo in `bblf` before)
 - updates examples of usage when reading content from the file (newlines were missing)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,11 @@
 
 This a Scala/JNI implementation of the [Babelfish](https://doc.bblf.sh/) client.
 It uses [ScalaPB](https://scalapb.github.io/grpc.html) for Protobuf/gRPC code
-generation and [libuast](https://github.com/bblfsh/libuast) for XPath queries. 
+generation and [libuast](https://github.com/bblfsh/libuast) for XPath queries.
+
+### Status
+
+Current `scala-clinet` v1.x only supports bblfsh protocol and UASTv1.
```
Comment (creachadair) [inline README.md]:
```suggestion
Current `scala-client` v1.x only supports bblfsh protocol and UASTv1.
```

File: src/main/scala/org/bblfsh/client/BblfshClient.scala [hunk 2]
```diff
@@ -1,44 +1,102 @@
...
     .usePlaintext(true)
     .maxInboundMessageSize(maxMsgSize)
     .build()
+
   private val stub = ProtocolServiceGrpc.blockingStub(channel)
 
-  def parse(name: String, content: String, lang: String = "", 
-            encoding: Encoding = Encoding.UTF8): ParseResponse = {
+  /**
+    *
```
Comment (creachadair) [inline src/main/scala/org/bblfsh/client/BblfshClient.scala]: (minor nit) Can this get a one-line summary?

File: src/main/scala/org/bblfsh/client/BblfshClient.scala [hunk 3]
```diff
@@ -1,44 +1,102 @@
...
+    lang: String = "",
+    encoding: Encoding = Encoding.UTF8
+  ): ParseResponse = parseWithOptions(name, content, lang, DEFAULT_TIMEOUT_SEC, encoding)
+
+  /**
+    * Parses the given file name and content,
+    * setting a bblfsh request timeout.
+    *
+    * Since v1.11, this API exposes the timeout.
+
```
Comment (creachadair) [inline src/main/scala/org/bblfsh/client/BblfshClient.scala]:
```suggestion
    *
```

File: src/test/scala/org/bblfsh/client/BblfshClientTimeoutTest.scala [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    val resp = client.parse(fileName, fileContent)
+    assert(!resp.errors.isEmpty)
+    assert(resp.errors(0).startsWith("rpc error: code = DeadlineExceeded desc"))
+  }
+
+  test("Parse requests with 1 min timeout works") {
+    val resp = client.parseWithTimeout(fileName, fileContent, 60)
+    assert(resp.errors.isEmpty)
+  }
+}
```
Comment (creachadair) [inline src/test/scala/org/bblfsh/client/BblfshClientTimeoutTest.scala]:
```suggestion
}

```

**Comment by bzz** [comment]
All feedback addressed in 0169dce

**Pull request merged**